fix(javascript): prevent import.meta.url replacement in web target

MIreland opened the pull request:
## Description
This PR fixes an issue where `import.meta.url` was unconditionally replaced by the local file path during the build, causing `file:///` URLs to appear in web builds.

## Fix
Modified `ImportMetaPlugin` to check if the target environment supports `document`. If it does (web target), `import.meta.url` is preserved in the output.

## Verification
Added a new test case `tests/rspack-test/configCases/parsing/import-meta-url-web` which asserts that `import.meta.url` is not replaced by a `file://` URL when targeting web.

MIreland committed:
fix(javascript): prevent import.meta.url replacement in web target

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/web-infra-dev/rspack?pullRequest=12266) <br/>All committers have signed the CLA.

chenjiahan commented on crates/rspack_plugin_javascript/src/parser_plugin/import_meta_plugin.rs:
```diff
@@ -79,11 +79,15 @@ impl JavascriptParserPlugin for ImportMetaPlugin {
-        start,
-        end,
-      ))
+      if parser.compiler_options.output.environment.supports_document() {
```
The script must be an ES module to use `import.meta.url`. We could not use `supports_document()` to determine this.

MIreland committed:
fix: check if module is ES module before using import.meta.url

The script must be an ES module to use import.meta.url. We cannot use supports_document() to determine this. This change adds a check for parser.is_esm before handling import.meta.url in both evaluate_identifier and member methods.

MIreland committed:
fix(javascript): preserve import.meta.url in destructuring for web targets

When destructuring import.meta in ES modules targeting web environments,
preserve import.meta.url using a getter function instead of replacing
it with a file:// URL. This ensures the browser can resolve it to the
actual script URL at runtime.

MIreland committed:
fix(test): update import-meta-url-web test to work with ES module output

- Add export statement to make test file an ES module
- Configure rspack to output ES module format (outputModule, module, chunkFormat)
- Update test expectations to verify import.meta.url is preserved
- Add test.config.js for moduleScope configuration

intellild commented:
Thanks. This would be a breaking change. Users may expect the original file path while preserve `import.meta.url` results in chunk serve url at runtime. A fine granted control is needed.
You can use `DefinePlugin` to workaround.

See the playground for detail 

https://playground.rspack.rs/#eyJyc3BhY2tWZXJzaW9uIjoiMi4xLjAtcmMuMCIsImlucHV0RmlsZXMiOlt7ImZpbGVuYW1lIjoicnNwYWNrLmNvbmZpZy5qcyIsInRleHQiOiJpbXBvcnQgKiBhcyByc3BhY2sgZnJvbSBcIkByc3BhY2svY29yZVwiXG5cbmV4cG9ydCBkZWZhdWx0IHtcbiAgbW9kZTogXCJkZXZlbG9wbWVudFwiLFxuICBkZXZ0b29sOiBmYWxzZSxcblx0ZW50cnk6IHtcblx0XHRtYWluOiBcIi4vaW5kZXguanNcIlxuXHR9LFxuICBvdXRwdXQ6IHtcbiAgICBtb2R1bGU6IHRydWUsXG4gIH0sXG4gIHBsdWdpbnM6IFtcbiAgICBuZXcgcnNwYWNrLkRlZmluZVBsdWdpbih7XG4gICAgICAnaW1wb3J0Lm1ldGEudXJsJzogSlNPTi5zdHJpbmdpZnkoJycpLFxuICAgICAgXCJfX2ltcG9ydF9tZXRhX19cIjogJ2ltcG9ydC5tZXRhJ1xuICAgIH0pXG4gIF0sXG59OyJ9LHsiZmlsZW5hbWUiOiJpbmRleC5qcyIsInRleHQiOiJjb25zb2xlLmxvZygnaW1wb3J0Lm1ldGEudXJsJywgaW1wb3J0Lm1ldGEudXJsKTtcblxuY29uc3QgeyB1cmwgfSA9IF9faW1wb3J0X21ldGFfXztcblxuY29uc29sZS5sb2coJ21ldGEnLCB1cmwpO1xuIn0seyJmaWxlbmFtZSI6ImxpYi5qcyIsInRleHQiOiJleHBvcnQgZGVmYXVsdCBcImxpYlwiOyJ9XX0=
